com.unity.render-pipelines.universal/Runtime/NativeRenderPass.cs

```diff
@@ -0,0 +1,528 @@
+                    AttachmentDescriptor currentAttachmentDescriptor;
+                    var usesTargetTexture = cameraData.targetTexture != null;
+                    var depthOnly = renderPass.depthOnly || (usesTargetTexture && cameraData.targetTexture.graphicsFormat == GraphicsFormat.DepthAuto);
+                    // Offscreen depth-only cameras need this set explicitly
+                    if (depthOnly && usesTargetTexture)
+                    {
+                        if (cameraData.targetTexture.graphicsFormat == GraphicsFormat.DepthAuto && !pass.overrideCameraTarget)
+                            passColorAttachment = new RenderTargetIdentifier(cameraData.targetTexture);
+                        else
+                            passColorAttachment = renderPass.colorAttachment;
+                        currentAttachmentDescriptor = new AttachmentDescriptor(GraphicsFormat.DepthAuto);
+                    }
+                    else
+                        currentAttachmentDescriptor =
+                            new AttachmentDescriptor(cameraData.cameraTargetDescriptor.graphicsFormat);
+
+                    if (pass.overrideCameraTarget)
+                        currentAttachmentDescriptor = new AttachmentDescriptor(pass.renderTargetFormat[0] != GraphicsFormat.None ? pass.renderTargetFormat[0] : GetDefaultGraphicsFormat(cameraData));
+
+                    var samples = pass.renderTargetSampleCount != -1
+                        ? pass.renderTargetSampleCount
+                        : cameraData.cameraTargetDescriptor.msaaSamples;
+
+                    var colorAttachmentTarget =
+                        (depthOnly || passColorAttachment != BuiltinRenderTextureType.CameraTarget)
+                        ? passColorAttachment : (usesTargetTexture
+                            ? new RenderTargetIdentifier(cameraData.targetTexture.colorBuffer)
+                            : BuiltinRenderTextureType.CameraTarget);
+
+                    var depthAttachmentTarget = (passDepthAttachment != BuiltinRenderTextureType.CameraTarget) ?
+                        passDepthAttachment : (usesTargetTexture
+                            ? new RenderTargetIdentifier(cameraData.targetTexture.depthBuffer)
+                            : BuiltinRenderTextureType.Depth);
+
+                    // TODO: review the lastPassToBB logic to mak it work with merged passes
+                    // keep track if this is the current camera's last pass and the RT is the backbuffer (BuiltinRenderTextureType.CameraTarget)
+                    // knowing isLastPassToBB can help decide the optimal store action as it gives us additional information about the current frame
+                    bool isLastPassToBB = pass.isLastPass && (colorAttachmentTarget == BuiltinRenderTextureType.CameraTarget);
+                    currentAttachmentDescriptor.ConfigureTarget(colorAttachmentTarget, ((uint)finalClearFlag & (uint)ClearFlag.Color) == 0, !(samples > 1 && isLastPassToBB));
```

Comment:
I know this proceeded this change, but is this actually the right logic? !(samples > 1 && isLastPassToBB)
Can you clarify it for me, so we want to store when no msaa or when this is not the last pass for the back buffer?

Reply:
this is old logic from before the merge. For now we always store but the next PR should optimize that

Reply:
in general though yes, if the pass wouldn't be merged (which was the case before this PR) this would try to discard if MSAA > 1 and is the last pass.
Now with the merging all the store/discard logic will have to take into account the mergeable pass list

Reply:
just to clarify: "why when MSAA > 1 we requested to store?"

this require an explanation of how Unity implicitly converts to resolve actions in the device implementations:

when MSAA is on, internally the device code converts DontCare actions in Resolve, and Store actions in StoreAndResolve. If the pass is the last pass to BB then we want to NOT store, so the action will become Resolve. Resolve still stores the resolved surface but will discards the MSAA'd surface, while StoreAndResolve would also store the MSAA'd surface which wastes a lot of bandwidth.

If MSAA is OFF: then you want to store the non-AA RT, otherwise the final result of the pass wouldn't make it to the BB memory as it would be discarded

Hope it makes sense now